chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomService.kt

```diff
@@ -78,33 +88,33 @@ internal class RoomService(
                     customData = customData,
                     userIds = userIds
             ).toJson()
-                    .flatMap { body -> client.doPost<Room>("/rooms", body) }
-                    .map { room ->
-                        roomStore += room
-                        userService.populateUserStore(room.memberUserIds)
-                        room
+                    .flatMap { body -> client.doPost<CreateRoomResponse>("/rooms", body) }
+                    .map { response ->
+                        roomStore.add(response)
+                        userService.populateUserStore(response.membership.userIds.toSet())
+                        mapToRoom(response.room, response.membership, null)
                     }
 
     fun deleteRoom(roomId: String): Result<String, Error> =
             legacyV2client.doDelete<Unit?>("/rooms/${URLEncoder.encode(roomId, "UTF-8")}")
                     .map {
-                        roomStore -= roomId
+                        roomStore.remove(roomId)
                         roomId
                     }
 
     fun leaveRoom(userId: String, roomId: String): Result<String, Error> =
             client.doPost<Unit?>("/users/${URLEncoder.encode(userId, "UTF-8")}/rooms/${URLEncoder.encode(roomId, "UTF-8")}/leave")
                     .map {
-                        roomStore -= roomId
+                        roomStore.remove(roomId)
                         roomId
                     }
 
     fun joinRoom(userId: String, roomId: String): Result<Room, Error> =
-            client.doPost<Room>("/users/${URLEncoder.encode(userId, "UTF-8")}/rooms/${URLEncoder.encode(roomId, "UTF-8")}/join")
-                    .map { room ->
-                        roomStore += room
-                        userService.populateUserStore(room.memberUserIds)
-                        room
+            client.doPost<JoinRoomResponse>("/users/${URLEncoder.encode(userId, "UTF-8")}/rooms/${URLEncoder.encode(roomId, "UTF-8")}/join")
+                    .map { response ->
+                        roomStore.add(response)
+                        userService.populateUserStore(response.membership.userIds.toSet())
+                        mapToRoom(response.room, response.membership, null)
```

Comment:
When we join a room shouldn't we get unread count?

Reply:
You can argue that we should, but IRL, we do not.
The unread count will follow immediately after in the subscription addedToRoom event, and we don't want to calculate it twice at considerable cost on the backend.